src/main/java/org/jenkinsci/plugins/GithubAuthenticationToken.java

```diff
@@ -329,22 +353,8 @@ public boolean hasRepositoryPermission(final String repositoryName) {
     }
 
     public boolean isPublicRepository(final String repositoryName) {
-        try {
-            return publicRepositoryCache.get(repositoryName,
-                new Callable<Boolean>() {
-                    @Override
-                    public Boolean call() throws Exception {
-                        GHRepository repository = loadRepository(repositoryName);
-                        // We don't have access so it must not be public (it could be non-existant)
-                        return repository != null && !repository.isPrivate();
-                    }
-                }
-            );
-        } catch (ExecutionException e) {
-            LOGGER.log(Level.SEVERE, "an exception was thrown", e);
-            throw new RuntimeException("authorization failed for user = "
-                    + getName(), e);
-        }
+        GHRepository repository = loadRepository(repositoryName);
```

Comment:
This is removing a cache; a performance feature with large Jenkins instances which need to interact with lots of repositories.  cc @i386

Reply:
Yes, it's removing the cache for public repositories, and is now instead using a public and private repository cache. Basically this only cached public repos before. I generalized it to try and cache any repository it tried to fetch.

Reply:
Ah, I guess I misunderstood that at face value.  Please bear with me as I'm not much of a Java coder (in my mind).